vault/counters.go

```diff
@@ -0,0 +1,152 @@
+package vault
+
+import (
+	"context"
+	"sort"
+	"sync/atomic"
+	"time"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/logical"
+)
+
+const requestCounterDatePathFormat = "2006/01"
+
+// RequestCounter stores the state of request counters for a single unspecified period.
+type RequestCounter struct {
+	// Total holds the sum total of all requests seen during the period.
+	// "All" does not include requests excluded by design, e.g. health checks and UI
+	// asset requests.
+	Total *uint64 `json:"total"`
+}
+
+// DatedRequestCounter holds request counters from a single period of time.
+type DatedRequestCounter struct {
+	// StartTime is when the period starts.
+	StartTime time.Time `json:"start_time"`
+	// RequestCounter counts requests.
+	RequestCounter
+}
+
+// AllRequestCounters contains all request counters from the dawn of time.
+type AllRequestCounters struct {
```

Comment:
Not certain this struct needs to exist, we can simply return an `[]DatedRequestCounter` from `loadAllRequestCounters`

Reply:
Ok.